presto-prometheus/src/test/java/io/prestosql/plugin/prometheus/TestPrometheusSplit.java

```diff
@@ -333,4 +334,83 @@ public void testGeneratedSplitTimesCorrectNonModuloZeroDurationToChunk()
                 "913600", "1000000");
         assertEquals(splitTimes, expectedSplitTimes);
     }
+
+    @Test
+    public void testGeneratedSplitTimesCorrectVersusMock()
+            throws Exception
+    {
+        String maxQueryRangeDurationStr = "120s";
+        String queryChunkSizeDurationStr = "30s";
+        long nowTimeMillis = mockNowTimeInMillis;
+        List<String> splitTimes = PrometheusSplitManager.generateTimesForSplits(nowTimeMillis, maxQueryRangeDurationStr, queryChunkSizeDurationStr);
+        List<String> promTimesReturned = mockPrometheusResponseToChunkedQueries(queryChunkSizeDurationStr, splitTimes);
+        assertEquals(convertMockTimesToStrings(promTimeValuesMock), promTimesReturned);
+    }
+
+    /**
+     * mock Prometheus chunked query responses (time values only)
+     *
+     * @param splitTimes         the end times that would be used for each Prometheus instant query
+     * @param queryChunkDuration the duration value that would be used for each query, `30s` for instance
+     * @return the values from the Prometheus data that would be return by all the chunked queries
+     */
+    private List<String> mockPrometheusResponseToChunkedQueries(String queryChunkDuration, List<String> splitTimes)
```

Comment:
I feel like we probably need a test for this method on it's own to make sure this produces the correct output.

Reply:
agreed, done in 9516dd0f